synapse/storage/databases/main/relations.py

```diff
@@ -355,17 +352,17 @@ async def get_applicable_edit(
                 original.event_id = relates_to_id
                 AND edit.type = original.type
                 AND edit.sender = original.sender
+                AND edit.room_id = original.room_id
```

Comment:
Is the point here that the DB already knows which room the original event was in, so we don't need to provide it ourselves?

(Just to make sure I understand more generally: is one event allowed to relate to another event in a different room?)

Reply:
> Is the point here that the DB already knows which room the original event was in, so we don't need to provide it ourselves?

Yeah, that's pretty much what we're taking advantage of. We want to do that so we can (potentially) query for edits to events across multiple rooms.

> (Just to make sure I understand more generally: is one event allowed to relate to another event in a different room?)

It is not allowed.